modules/questions.py

```diff
@@ -32,226 +27,418 @@
             return Response(
                 confidence=8,
-                callback=self.post_count_questions_message,
-                args=[match, message],
-                why="I was asked to count questions",
+                callback=self.message_post_question,
+                args=[query, message],
+                why="I was asked for next questions",
             )
-        if match := self.re_next_question.search(text):
+        if query := QuestionInfoQuery.parse(text, self.last_question_id):
             return Response(
                 confidence=8,
-                callback=self.post_next_question_messages,
-                args=[match, message],
-                why="I was asked for next questions",
+                callback=self.message_get_question_info,
+                args=[query, message],
+                why="I was asked to post info about a message",
             )
-        return Response(why="Left QuestionManager without matching to any Regex")
-
-    def get_questions_df(self, qq: QuestionQuery) -> pd.DataFrame:
-        """Get questions from with `status="Not started"`"""
-        request_res = self.send_coda_questions_request(qq)
-        rows = [self.parse_row(row) for row in request_res["items"]]
-        return pd.DataFrame(rows)
-
-    def parse_row(self, row: dict) -> ParsedRow:
-        """Parse row from "All answers" table"""
-        title = row["values"]["Edit Answer"]
-        url = row["values"]["Link"]
-        status = row["values"]["Status"]
-        tags = row["values"]["Tags"].split(",")
-        last_asked_on_discord = self.adjust_date(row["values"]["Last Asked On Discord"])
-        return {
-            "id": row["id"],
-            "title": title,
-            "url": url,
-            "status": status,
-            "tags": tags,
-            "last_asked_on_discord": last_asked_on_discord,
-        }
+        if query := SetQuestionQuery.parse(text, self.last_question_id):
+            return Response(
+                confidence=8,
+                callback=self.message_set_question,
+                args=[query, message],
+                why=f"I was asked to set status of question `{query.id}` to `{query.status}`",
+            )
+        return Response(why="Left QuestionManager without matching to any query")
+
+    ##################
+    # Review request #
+    ##################
+
+    def is_review_request(self, message: ServiceMessage) -> Optional[SetQuestionQuery]:
+        """Is this message a review request with link do GDoc?"""
+        text = message.content
+        if (gdoc_link := parse_gdoc_link(text)) is None:
+            return
+        if (question := self.get_question_by_gdoc_link(gdoc_link)) is None:
+            return
+
+        if "@reviewer" in text:
+            new_status = "In review"
+        elif "@feedback-sketch" in text:
+            new_status = "Bulletpoint sketch"
+        elif "@feedback" in text:
+            new_status = "In progress"
```

Comment:
@plexish is there some documentation that specifies the workflow that we should keep in code comments for future maintainers?

Reply:
there's this log of me describing it and inventing it

> MatthewBaggins — Today at 14:06
to which role's at-mention in #editing should Stampy react by setting question's status to "In review"?
should be @reviewer, @feedback, @editor, or which two/all of them?
plex — Today at 14:10
setting in review should not be gated, everyone should be able to do it
it's just setting on site that should be gated, and tag exceptional 
MatthewBaggins — Today at 14:11
I'm not talking about checking user's role
plex — Today at 14:11
oh I see
just reviewer
MatthewBaggins — Today at 14:11
ok
plex — Today at 14:12
@ feedback  should if the question Is not started move it to in progress
MatthewBaggins — Today at 14:12
ok
and this should work only in #editing?
plex — Today at 14:13
no
MatthewBaggins — Today at 14:13
or all channels or some subset of them?
plex — Today at 14:13
all channels
MatthewBaggins — Today at 14:13
ok
plex — Today at 14:14
if the question is bulletpoint sketch this shouldn't move it to in progress though
MatthewBaggins — Today at 14:14
got it
plex — Today at 14:15
or actually maybe I should make a new roll for bullet. sketch
I'm going to avoid correcting stuff as much, my speech recognition software is not perfect
but it's mostly understandable
I think I might make another roll for bullet. sketch review
which would mean that asking for feedback always makes it in progress
asking for sketch feedback always makes it bullet . sketch

documentation could be something like:
When someone asks for @feedback-sketch with the gdoc URL, set status to Bulletpoint sketch. When they ask for @feedback, set status to In progress. When someone asks for @review set status to review.